Kragle - to automate and verify projects are ready

adean20csc opened the pull request:
There are a couple of commits tied together here.

prep-projects role - to copy files from the /opt/autodeploy/ staging area to the project folders
verify-files - to verify everything exists in the /opt/autodeploy/ staging area


[root@nuc ansible]# ansible-playbook main.yml

PLAY [Setup kragle] ***********************************************************

TASK: [base | install updates] ************************************************
ok: [localhost]

TASK: [base | install support packages] ***************************************
ok: [localhost] => (item=bridge-utils,dhcp,docker,ipmitool,iptables-services,ntp,python-passlib,python-pip,sshpass,vim,wget,httpd,createrepo,yum-utils,unzip)

TASK: [base | install support packages with pip] ******************************
skipping: [localhost] => (item=docker-py)
skipping: [localhost] => (item=selenium)

TASK: [base | install docker-py when offline] *********************************
ok: [localhost]

TASK: [base | create .ssh directory] ******************************************
ok: [localhost]

TASK: [base | Create SSH Key Pair for root user] ******************************
ok: [localhost]

TASK: [base | copy .ssh/config] ***********************************************
ok: [localhost]

TASK: [base | Retrieve deployment node public key] ****************************
changed: [localhost]

TASK: [base | Add deployment node SSH key to node authorized users] ***********
ok: [localhost]

TASK: [base | disable firewalld] **********************************************
ok: [localhost]

TASK: [base | create directories for hanlon and scaleio] **********************
ok: [localhost] => (item=/opt/hanlon)
ok: [localhost] => (item=/opt/hanlon/image)
ok: [localhost] => (item=/opt/deploy)
ok: [localhost] => (item=/opt/autodeploy/resources/scaleio)

TASK: [base | Set the correct timezone] ***************************************
ok: [localhost]

TASK: [base | Configure NTP] **************************************************
ok: [localhost]

TASK: [base | copy iptables config file] **************************************
ok: [localhost]

TASK: [base | start & enable services] ****************************************
ok: [localhost] => (item=iptables)
ok: [localhost] => (item=ntpd)
ok: [localhost] => (item=docker)

TASK: [base | get latest tags] ************************************************
skipping: [localhost] => (item={'repo': 'git://github.com/csc/slimer', 'tag': 'https://api.github.com/repos/csc/slimer/tags', 'name': 'slimer'})
skipping: [localhost] => (item={'repo': 'git://github.com/csc/kragle', 'tag': 'https://api.github.com/repos/csc/kragle/tags', 'name': 'kragle'})
skipping: [localhost] => (item={'repo': 'git://github.com/csc/ansible-scaleio', 'tag': 'https://api.github.com/repos/csc/ansible-scaleio/tags', 'name': 'ansible-scaleio'})

TASK: [base | clone repos from git] *******************************************
skipping: [localhost] => (item={u'skipped': True, u'changed': False})
skipping: [localhost] => (item={u'skipped': True, u'changed': False})
skipping: [localhost] => (item={u'skipped': True, u'changed': False})

TASK: [createrepo | create repo directory for httpd] **************************
ok: [localhost]

TASK: [createrepo | copy repo files to http directory] ************************
changed: [localhost]

TASK: [createrepo | createrepo] ***********************************************
changed: [localhost]

TASK: [createrepo | repofile to provide to node] ******************************
ok: [localhost]

TASK: [createrepo | start httpd] **********************************************
changed: [localhost]

TASK: [prep-projects | Check if ScaleIO files exist] **************************
ok: [localhost]

TASK: [prep-projects | Download ScaleIO files for ansible-scaleio project] ****
skipping: [localhost]

TASK: [prep-projects | Extract the ScaleIO source files] **********************
changed: [localhost]

TASK: [prep-projects | Find ScaleIO RPMs to be copied] ************************
changed: [localhost]

TASK: [prep-projects | Copy ScaleIO RPMs to the ansible-scaleio project] ******
ok: [localhost] => (item=/opt/autodeploy/resources/scaleio/ScaleIO_1.32.2_Gateway_for_Linux_Download/EMC-ScaleIO-gateway-1.32-2451.4.noarch.rpm)
ok: [localhost] => (item=/opt/autodeploy/resources/scaleio/ScaleIO_1.32.2_RHEL7_Download/EMC-ScaleIO-callhome-1.32-2451.4.el7.x86_64.rpm)
ok: [localhost] => (item=/opt/autodeploy/resources/scaleio/ScaleIO_1.32.2_RHEL7_Download/EMC-ScaleIO-lia-1.32-2451.4.el7.x86_64.rpm)
ok: [localhost] => (item=/opt/autodeploy/resources/scaleio/ScaleIO_1.32.2_RHEL7_Download/EMC-ScaleIO-mdm-1.32-2451.4.el7.x86_64.rpm)
ok: [localhost] => (item=/opt/autodeploy/resources/scaleio/ScaleIO_1.32.2_RHEL7_Download/EMC-ScaleIO-sdc-1.32-2451.4.el7.x86_64.rpm)
ok: [localhost] => (item=/opt/autodeploy/resources/scaleio/ScaleIO_1.32.2_RHEL7_Download/EMC-ScaleIO-sds-1.32-2451.4.el7.x86_64.rpm)
ok: [localhost] => (item=/opt/autodeploy/resources/scaleio/ScaleIO_1.32.2_RHEL7_Download/EMC-ScaleIO-tb-1.32-2451.4.el7.x86_64.rpm)

TASK: [prep-projects | Find ScaleIO Openstack driver file to extract] *********
changed: [localhost]

TASK: [prep-projects | Extract the ScaleIO OpenStack drivers] *****************
changed: [localhost] => (item=/opt/autodeploy/resources/scaleio/ScaleIO_1.32.2_OpenStack_Driver_Download/EMC-ScaleIO-openstack-driver-1.32-2451.4-juno.zip)

TASK: [prep-projects | Copy ScaleIO OpenStack-Compute Drivers to ansible-scaleio project] ***
ok: [localhost] => (item=scaleio.filters)
ok: [localhost] => (item=scaleio.py)
ok: [localhost] => (item=scaleio_config.py)
ok: [localhost] => (item=scaleio_install.py)
ok: [localhost] => (item=scaleiolibvirtdriver.py)

TASK: [prep-projects | Copy ScaleIO OpenStack-Controller Drivers to ansible-scaleio project] ***
ok: [localhost] => (item=scaleio.filters)
ok: [localhost] => (item=scaleio.py)
ok: [localhost] => (item=scaleio_config.py)
ok: [localhost] => (item=scaleio_install.py)
ok: [localhost] => (item=scaleiolibvirtdriver.py)

TASK: [prep-projects | Check if RHEL minimized image is in the resources directory] ***
skipping: [localhost]

TASK: [prep-projects | Check if RHEL minimized image is in Hanlon image directory] ***
skipping: [localhost]

TASK: [prep-projects | Copy the RHEL minimized image when offline] ************
skipping: [localhost]

PLAY RECAP ********************************************************************
localhost                  : ok=30   changed=8    unreachable=0    failed=0

mtnbikenc commented on ansible/inventory/group_vars/all.yml:
```diff
@@ -1,17 +1,29 @@
-auto_deploy_node: 172.17.16.10
-hanlon_base_url: http://{{ auto_deploy_node }}:8026/hanlon/api/v1/
-dns1: 192.168.70.3
+autodeploynode: 192.168.111.199
```
For project consistency sake, I'd like to see `auto_deploy_node`

mtnbikenc commented on ansible/inventory/group_vars/all.yml:
```diff
@@ -1,17 +1,29 @@
+docker_path: "{{ resources_base_path }}/docker/"
+rpm_file: final-rpm-list.txt
+rpm_path: "{{ resources_base_path }}/rpms/"
+rpm_list: "{{ projects_base_path }}/wiley/files/{{ rpm_file }}"
```
Not sure I understand why the rpm file list is under wiley.

adean20csc replied:
because it was created in the wiley/docs, I asked about moving it to wiley/files since it is used by the project and not needed in the docs

mtnbikenc replied:
Ok since wiley does not install any rpms and kragle is where the offline repo is located, the file should be under kragle/files instead.  Only kragle and slimer install RPMs and I think that is why I was wondering about wiley.  And I agree that it should not be in the wiley repo either.

jcpowermac commented on ansible/roles/prep-projects/tasks/wiley.yml:
```diff
@@ -0,0 +1,21 @@
+---
```
@adean20csc as discussed in chat, add a task to create the minimized iso.

adean20csc replied:
This will be added in a diff PR

mtnbikenc commented on ansible/inventory/group_vars/all.yml:
```diff
@@ -30,3 +45,46 @@ git_repos:
+  - slimer
+  - ansible-scaleio
+
+iso_files:
```
Can a variables be used here so these file names don't have to be typed in twice, once above, and once here?

adean20csc replied:
added these variables

mtnbikenc commented on ansible/inventory/group_vars/all.yml:
```diff
@@ -19,7 +31,10 @@ pxe_dhcp_range_start: 172.17.16.20
 pxe_dhcp_range_end: 172.17.16.50
 domain_name: lordbusiness
 
+rhel_iso_image: rhel-server-7.1-x86_64-dvd.iso
```
I'm not finding where this variable (rhel_iso_image )is used.

adean20csc replied:
It is used as a variable

mtnbikenc commented:
I would prefer the extra files created by `ansible-galaxy init <rolename>` be removed unless they are being changed from the default boilerplate.

jcpowermac commented on ansible/main.yml:
```diff
@@ -4,5 +4,5 @@
     - base
-    - { role: createrepo, when: offline}
-  
+    - { role: createrepo, when: offline }
```
@adean20csc missing the role verify-files?

adean20csc replied:
Not using it at this point

adean20csc commented:
PLAY [Setup kragle] ***********************************************************

TASK: [verify-files | Check if source project files exist on autodeploynode] ***
ok: [localhost] => (item=kragle)
ok: [localhost] => (item=wiley)
ok: [localhost] => (item=slimer)
ok: [localhost] => (item=ansible-scaleio)

TASK: [verify-files | Copy source project files to autodeploynode when offline] ***
skipping: [localhost] => (item={'invocation': {'module_name': u'stat', 'module_args': ''}, 'item': 'kragle', u'stat': {u'uid': 1000, u'exists': True, u'woth': False, u'mtime': 1444959413.6951683, u'inode': 7977571, u'isgid': False, u'size': 4096, u'wgrp': False, u'isuid': False, u'isreg': False, u'pw_name': u'kragle', u'gid': 0, u'ischr': False, u'wusr': True, u'xoth': True, u'rusr': True, u'nlink': 8, u'issock': False, u'rgrp': True, u'gr_name': u'root', u'path': u'/opt/autodeploy/projects/kragle', u'xusr': True, u'atime': 1445342207.4715219, u'isdir': True, u'ctime': 1445344428.3137386, u'isblk': False, u'xgrp': True, u'dev': 64769, u'roth': True, u'isfifo': False, u'mode': u'0755', u'islnk': False}, u'changed': False})
skipping: [localhost] => (item={'invocation': {'module_name': u'stat', 'module_args': ''}, 'item': 'wiley', u'stat': {u'uid': 1000, u'exists': True, u'woth': False, u'mtime': 1444972888.9114833, u'inode': 67109624, u'isgid': False, u'size': 4096, u'wgrp': False, u'isuid': False, u'isreg': False, u'pw_name': u'kragle', u'gid': 0, u'ischr': False, u'wusr': True, u'xoth': True, u'rusr': True, u'nlink': 9, u'issock': False, u'rgrp': True, u'gr_name': u'root', u'path': u'/opt/autodeploy/projects/wiley', u'xusr': True, u'atime': 1445344704.2997656, u'isdir': True, u'ctime': 1445344428.3157387, u'isblk': False, u'xgrp': True, u'dev': 64769, u'roth': True, u'isfifo': False, u'mode': u'0755', u'islnk': False}, u'changed': False})
skipping: [localhost] => (item={'invocation': {'module_name': u'stat', 'module_args': ''}, 'item': 'slimer', u'stat': {u'uid': 1000, u'exists': True, u'woth': False, u'mtime': 1444959414.0001683, u'inode': 67109528, u'isgid': False, u'size': 4096, u'wgrp': False, u'isuid': False, u'isreg': False, u'pw_name': u'kragle', u'gid': 0, u'ischr': False, u'wusr': True, u'xoth': True, u'rusr': True, u'nlink': 7, u'issock': False, u'rgrp': True, u'gr_name': u'root', u'path': u'/opt/autodeploy/projects/slimer', u'xusr': True, u'atime': 1445344428.3137386, u'isdir': True, u'ctime': 1445344428.3157387, u'isblk': False, u'xgrp': True, u'dev': 64769, u'roth': True, u'isfifo': False, u'mode': u'0755', u'islnk': False}, u'changed': False})
skipping: [localhost] => (item={'invocation': {'module_name': u'stat', 'module_args': ''}, 'item': 'ansible-scaleio', u'stat': {u'uid': 1000, u'exists': True, u'woth': False, u'mtime': 1444959413.5571682, u'inode': 136953041, u'isgid': False, u'size': 4096, u'wgrp': False, u'isuid': False, u'isreg': False, u'pw_name': u'kragle', u'gid': 0, u'ischr': False, u'wusr': True, u'xoth': True, u'rusr': True, u'nlink': 5, u'issock': False, u'rgrp': True, u'gr_name': u'root', u'path': u'/opt/autodeploy/projects/ansible-scaleio', u'xusr': True, u'atime': 1445344428.3127387, u'isdir': True, u'ctime': 1445344428.3137386, u'isblk': False, u'xgrp': True, u'dev': 64769, u'roth': True, u'isfifo': False, u'mode': u'0755', u'islnk': False}, u'changed': False})

TASK: [verify-files | Check if source ISO files exist on autodeploynode] ******
ok: [localhost] => (item=hnl_mk_debug-image.2.0.1.iso)
ok: [localhost] => (item=rhel-server-7.1-x86_64-dvd.iso)
ok: [localhost] => (item=rhel-server-7.1-x86_64-dvd-hnl.iso)
ok: [localhost] => (item=rhel-guest-image-7.1-20150224.0.x86_64.qcow2)

TASK: [verify-files | Copy source ISO files to autodeploynode when offline] ***
skipping: [localhost] => (item={'invocation': {'module_name': u'stat', 'module_args': ''}, 'item': u'hnl_mk_debug-image.2.0.1.iso', u'stat': {u'uid': 1000, u'exists': True, u'woth': False, u'mtime': 1444939206.8611965, u'inode': 202124560, u'isgid': False, u'size': 54145024, u'wgrp': False, u'isuid': False, u'isreg': True, u'pw_name': u'kragle', u'gid': 0, u'ischr': False, u'wusr': True, u'xoth': False, u'rusr': True, u'nlink': 1, u'issock': False, u'rgrp': True, u'gr_name': u'root', u'path': u'/opt/autodeploy/resources/ISO/hnl_mk_debug-image.2.0.1.iso', u'xusr': False, u'atime': 1445355461.3078153, u'md5': u'b61423d24f4bda7e64b04163493fc0d6', u'isdir': False, u'ctime': 1445344428.3127387, u'isblk': False, u'xgrp': False, u'dev': 64769, u'roth': True, u'isfifo': False, u'mode': u'0644', u'checksum': u'4ed8350e7d332c9a8e65cd46041a2e5b72cf60b1', u'islnk': False}, u'changed': False})
---
skipping: [localhost] => (item={'invocation': {'module_name': u'stat', 'module_args': ''}, 'item': u'rhel-server-7.1-x86_64-dvd.iso', u'stat': {u'uid': 1000, u'exists': True, u'woth': False, u'mtime': 1444939178.3231938, u'inode': 202124558, u'isgid': False, u'size': 3890216960, u'wgrp': False, u'isuid': False, u'isreg': True, u'pw_name': u'kragle', u'gid': 0, u'ischr': False, u'wusr': True, u'xoth': False, u'rusr': True, u'nlink': 1, u'issock': False, u'rgrp': True, u'gr_name': u'root', u'path': u'/opt/autodeploy/resources/ISO/rhel-server-7.1-x86_64-dvd.iso', u'xusr': False, u'atime': 1445344623.6887577, u'md5': u'370662c78e36528bbaf3f2b901c29d6f', u'isdir': False, u'ctime': 1445344428.3127387, u'isblk': False, u'xgrp': False, u'dev': 64769, u'roth': True, u'isfifo': False, u'mode': u'0644', u'checksum': u'9af515956bb428f697aa6e30791c032a00060d1e', u'islnk': False}, u'changed': False})
skipping: [localhost] => (item={'invocation': {'module_name': u'stat', 'module_args': ''}, 'item': u'rhel-server-7.1-x86_64-dvd-hnl.iso', u'stat': {u'uid': 0, u'exists': True, u'woth': False, u'mtime': 1445346799.33697, u'inode': 202131250, u'isgid': False, u'size': 1212985344, u'wgrp': False, u'isuid': False, u'isreg': True, u'pw_name': u'root', u'gid': 0, u'ischr': False, u'wusr': True, u'xoth': False, u'rusr': True, u'nlink': 1, u'issock': False, u'rgrp': True, u'gr_name': u'root', u'path': u'/opt/autodeploy/resources/ISO/rhel-server-7.1-x86_64-dvd-hnl.iso', u'xusr': False, u'atime': 1445355476.3648167, u'md5': u'212e9e6832399b14a0511c8d2bf91cd0', u'isdir': False, u'ctime': 1445346799.33697, u'isblk': False, u'xgrp': False, u'dev': 64769, u'roth': True, u'isfifo': False, u'mode': u'0644', u'checksum': u'dbd8b6c329052cc60a199dd810f7040747729cff', u'islnk': False}, u'changed': False})
skipping: [localhost] => (item={'invocation': {'module_name': u'stat', 'module_args': ''}, 'item': u'rhel-guest-image-7.1-20150224.0.x86_64.qcow2', u'stat': {u'uid': 1000, u'exists': True, u'woth': False, u'mtime': 1444939203.785196, u'inode': 202124559, u'isgid': False, u'size': 425956864, u'wgrp': False, u'isuid': False, u'isreg': True, u'pw_name': u'kragle', u'gid': 0, u'ischr': False, u'wusr': True, u'xoth': False, u'rusr': True, u'nlink': 1, u'issock': False, u'rgrp': True, u'gr_name': u'root', u'path': u'/opt/autodeploy/resources/ISO/rhel-guest-image-7.1-20150224.0.x86_64.qcow2', u'xusr': False, u'atime': 1445355481.042817, u'md5': u'b068d0e9531699516174a436bf2c300c', u'isdir': False, u'ctime': 1445344428.3127387, u'isblk': False, u'xgrp': False, u'dev': 64769, u'roth': True, u'isfifo': False, u'mode': u'0644', u'checksum': u'5f3d607e7a490ae99dbdbb85ae979122ed5567fb', u'islnk': False}, u'changed': False})

TASK: [verify-files | Check if source docker files exist on autodeploynode] ***
ok: [localhost] => (item=cscdock_hanlon.tar)
ok: [localhost] => (item=cscdock_atftpd.tar)
ok: [localhost] => (item=mongo.tar)

TASK: [verify-files | Copy source docker files to autodeploynode when offline] ***
skipping: [localhost] => (item={'invocation': {'module_name': u'stat', 'module_args': ''}, 'item': 'cscdock_hanlon.tar', u'stat': {u'uid': 1000, u'exists': True, u'woth': False, u'mtime': 1444939284.045204, u'inode': 2344562, u'isgid': False, u'size': 737451520, u'wgrp': False, u'isuid': False, u'isreg': True, u'pw_name': u'kragle', u'gid': 0, u'ischr': False, u'wusr': True, u'xoth': False, u'rusr': True, u'nlink': 1, u'issock': False, u'rgrp': True, u'gr_name': u'root', u'path': u'/opt/autodeploy/resources/docker/cscdock_hanlon.tar', u'xusr': False, u'atime': 1445355482.7408173, u'md5': u'e15e5307a59ac256d86e08fd2d627049', u'isdir': False, u'ctime': 1445344428.3127387, u'isblk': False, u'xgrp': False, u'dev': 64769, u'roth': True, u'isfifo': False, u'mode': u'0644', u'checksum': u'aa6eade25f96eaffff5fcd3610b99a5bcc0671bc', u'islnk': False}, u'changed': False})
skipping: [localhost] => (item={'invocation': {'module_name': u'stat', 'module_args': ''}, 'item': 'cscdock_atftpd.tar', u'stat': {u'uid': 1000, u'exists': True, u'woth': False, u'mtime': 1444939283.564204, u'inode': 2344560, u'isgid': False, u'size': 210205184, u'wgrp': False, u'isuid': False, u'isreg': True, u'pw_name': u'kragle', u'gid': 0, u'ischr': False, u'wusr': True, u'xoth': False, u'rusr': True, u'nlink': 1, u'issock': False, u'rgrp': True, u'gr_name': u'root', u'path': u'/opt/autodeploy/resources/docker/cscdock_atftpd.tar', u'xusr': False, u'atime': 1445355485.5838177, u'md5': u'cbec0031f66d6740dfd906040d732176', u'isdir': False, u'ctime': 1445344428.3127387, u'isblk': False, u'xgrp': False, u'dev': 64769, u'roth': True, u'isfifo': False, u'mode': u'0644', u'checksum': u'6178c010cd2190ce118f7cb91c3e6263adf7c8ad', u'islnk': False}, u'changed': False})
skipping: [localhost] => (item={'invocation': {'module_name': u'stat', 'module_args': ''}, 'item': 'mongo.tar', u'stat': {u'uid': 1000, u'exists': True, u'woth': False, u'mtime': 1445018430.2809272, u'inode': 2344561, u'isgid': False, u'size': 267695616, u'wgrp': False, u'isuid': False, u'isreg': True, u'pw_name': u'kragle', u'gid': 0, u'ischr': False, u'wusr': True, u'xoth': False, u'rusr': True, u'nlink': 1, u'issock': False, u'rgrp': True, u'gr_name': u'root', u'path': u'/opt/autodeploy/resources/docker/mongo.tar', u'xusr': False, u'atime': 1445355486.4398177, u'md5': u'd9488bf99a1a04437bea0f64e884c7dd', u'isdir': False, u'ctime': 1445344428.3127387, u'isblk': False, u'xgrp': False, u'dev': 64769, u'roth': True, u'isfifo': False, u'mode': u'0644', u'checksum': u'4466db0a806dad533aa48ed9b1fea796f2e0f5ea', u'islnk': False}, u'changed': False})

PLAY RECAP ********************************************************************
localhost                  : ok=6    changed=0    unreachable=0    failed=0

[root@nuc ansible]# vi main.yml
[root@nuc ansible]# ansible-playbook main.yml

PLAY [Setup kragle] ***********************************************************

TASK: [base | install updates] ************************************************
ok: [localhost]

TASK: [base | install support packages] ***************************************
ok: [localhost] => (item=bridge-utils,dhcp,docker,ipmitool,iptables-services,ntp,python-passlib,sshpass,vim,httpd)

TASK: [base | install support packages with pip] ******************************
skipping: [localhost] => (item=docker-py)
skipping: [localhost] => (item=selenium)

TASK: [base | install docker-py when offline] *********************************
ok: [localhost]

TASK: [base | create .ssh directory] ******************************************
ok: [localhost]

TASK: [base | Create SSH Key Pair for root user] ******************************
ok: [localhost]

TASK: [base | copy .ssh/config] ***********************************************
ok: [localhost]

TASK: [base | Retrieve deployment node public key] ****************************
changed: [localhost]

TASK: [base | Add deployment node SSH key to node authorized users] ***********
ok: [localhost]

TASK: [base | disable firewalld] **********************************************
ok: [localhost]

TASK: [base | create directories for hanlon and scaleio] **********************
ok: [localhost] => (item=/opt/hanlon)
ok: [localhost] => (item=/opt/hanlon/image)
ok: [localhost] => (item=/opt/deploy)
ok: [localhost] => (item=/opt/autodeploy/resources/scaleio)

TASK: [base | Set the correct timezone] ***************************************
ok: [localhost]

TASK: [base | Configure NTP] **************************************************
ok: [localhost]

TASK: [base | copy iptables config file] **************************************
ok: [localhost]

TASK: [base | start & enable services] ****************************************
ok: [localhost] => (item=iptables)
ok: [localhost] => (item=ntpd)
ok: [localhost] => (item=docker)

TASK: [base | get latest tags] ************************************************
skipping: [localhost] => (item={'repo': 'git://github.com/csc/slimer', 'tag': 'https://api.github.com/repos/csc/slimer/tags', 'name': 'slimer'})
skipping: [localhost] => (item={'repo': 'git://github.com/csc/kragle', 'tag': 'https://api.github.com/repos/csc/kragle/tags', 'name': 'kragle'})
skipping: [localhost] => (item={'repo': 'git://github.com/csc/ansible-scaleio', 'tag': 'https://api.github.com/repos/csc/ansible-scaleio/tags', 'name': 'ansible-scaleio'})

TASK: [base | clone repos from git] *******************************************
skipping: [localhost] => (item={u'skipped': True, u'changed': False})
skipping: [localhost] => (item={u'skipped': True, u'changed': False})
skipping: [localhost] => (item={u'skipped': True, u'changed': False})

TASK: [createrepo | create repo directory for httpd] **************************
ok: [localhost]

TASK: [createrepo | copy repo files to http directory] ************************
changed: [localhost]

TASK: [createrepo | createrepo] ***********************************************
changed: [localhost]

TASK: [createrepo | repofile to provide to node] ******************************
ok: [localhost]

TASK: [createrepo | start httpd] **********************************************
changed: [localhost]

TASK: [prep-projects | Check if ScaleIO files exist] **************************
ok: [localhost]

TASK: [prep-projects | Download ScaleIO files for ansible-scaleio project] ****
skipping: [localhost]

TASK: [prep-projects | Extract the ScaleIO source files] **********************
changed: [localhost]

TASK: [prep-projects | Find ScaleIO RPMs to be copied] ************************
changed: [localhost]

TASK: [prep-projects | Copy ScaleIO RPMs to the ansible-scaleio project] ******
ok: [localhost] => (item=/opt/autodeploy/resources/scaleio/ScaleIO_1.32.2_Gateway_for_Linux_Download/EMC-ScaleIO-gateway-1.32-2451.4.noarch.rpm)
ok: [localhost] => (item=/opt/autodeploy/resources/scaleio/ScaleIO_1.32.2_RHEL7_Download/EMC-ScaleIO-callhome-1.32-2451.4.el7.x86_64.rpm)
ok: [localhost] => (item=/opt/autodeploy/resources/scaleio/ScaleIO_1.32.2_RHEL7_Download/EMC-ScaleIO-lia-1.32-2451.4.el7.x86_64.rpm)
ok: [localhost] => (item=/opt/autodeploy/resources/scaleio/ScaleIO_1.32.2_RHEL7_Download/EMC-ScaleIO-mdm-1.32-2451.4.el7.x86_64.rpm)
ok: [localhost] => (item=/opt/autodeploy/resources/scaleio/ScaleIO_1.32.2_RHEL7_Download/EMC-ScaleIO-sdc-1.32-2451.4.el7.x86_64.rpm)
ok: [localhost] => (item=/opt/autodeploy/resources/scaleio/ScaleIO_1.32.2_RHEL7_Download/EMC-ScaleIO-sds-1.32-2451.4.el7.x86_64.rpm)
ok: [localhost] => (item=/opt/autodeploy/resources/scaleio/ScaleIO_1.32.2_RHEL7_Download/EMC-ScaleIO-tb-1.32-2451.4.el7.x86_64.rpm)

TASK: [prep-projects | Find ScaleIO Openstack driver file to extract] *********
changed: [localhost]

TASK: [prep-projects | Extract the ScaleIO OpenStack drivers] *****************
changed: [localhost] => (item=/opt/autodeploy/resources/scaleio/ScaleIO_1.32.2_OpenStack_Driver_Download/EMC-ScaleIO-openstack-driver-1.32-2451.4-juno.zip)

TASK: [prep-projects | Copy ScaleIO OpenStack-Compute Drivers to ansible-scaleio project] ***
ok: [localhost] => (item=scaleio.filters)
ok: [localhost] => (item=scaleio.py)
ok: [localhost] => (item=scaleio_config.py)
ok: [localhost] => (item=scaleio_install.py)
ok: [localhost] => (item=scaleiolibvirtdriver.py)

TASK: [prep-projects | Copy ScaleIO OpenStack-Controller Drivers to ansible-scaleio project] ***
ok: [localhost] => (item=scaleio.filters)
ok: [localhost] => (item=scaleio.py)
ok: [localhost] => (item=scaleio_config.py)
ok: [localhost] => (item=scaleio_install.py)
ok: [localhost] => (item=scaleiolibvirtdriver.py)

TASK: [prep-projects | Check if RHEL minimized image is in the resources directory] ***
skipping: [localhost]

TASK: [prep-projects | Create the Hanlon RHEL minimized ISO file] *************
skipping: [localhost]

TASK: [prep-projects | Check if RHEL minimized image is in Hanlon image directory] ***
skipping: [localhost]

TASK: [prep-projects | Copy the RHEL minimized image when offline] ************
skipping: [localhost]

TASK: [verify-files | Check if source project files exist on autodeploynode] ***
ok: [localhost] => (item=kragle)
ok: [localhost] => (item=wiley)
ok: [localhost] => (item=slimer)
ok: [localhost] => (item=ansible-scaleio)

TASK: [verify-files | Copy source project files to autodeploynode when offline] ***
skipping: [localhost] => (item={'invocation': {'module_name': u'stat', 'module_args': ''}, 'item': 'kragle', u'stat': {u'uid': 1000, u'exists': True, u'woth': False, u'mtime': 1444959413.6951683, u'inode': 7977571, u'isgid': False, u'size': 4096, u'wgrp': False, u'isuid': False, u'isreg': False, u'pw_name': u'kragle', u'gid': 0, u'ischr': False, u'wusr': True, u'xoth': True, u'rusr': True, u'nlink': 8, u'issock': False, u'rgrp': True, u'gr_name': u'root', u'path': u'/opt/autodeploy/projects/kragle', u'xusr': True, u'atime': 1445342207.4715219, u'isdir': True, u'ctime': 1445344428.3137386, u'isblk': False, u'xgrp': True, u'dev': 64769, u'roth': True, u'isfifo': False, u'mode': u'0755', u'islnk': False}, u'changed': False})
skipping: [localhost] => (item={'invocation': {'module_name': u'stat', 'module_args': ''}, 'item': 'wiley', u'stat': {u'uid': 1000, u'exists': True, u'woth': False, u'mtime': 1444972888.9114833, u'inode': 67109624, u'isgid': False, u'size': 4096, u'wgrp': False, u'isuid': False, u'isreg': False, u'pw_name': u'kragle', u'gid': 0, u'ischr': False, u'wusr': True, u'xoth': True, u'rusr': True, u'nlink': 9, u'issock': False, u'rgrp': True, u'gr_name': u'root', u'path': u'/opt/autodeploy/projects/wiley', u'xusr': True, u'atime': 1445344704.2997656, u'isdir': True, u'ctime': 1445344428.3157387, u'isblk': False, u'xgrp': True, u'dev': 64769, u'roth': True, u'isfifo': False, u'mode': u'0755', u'islnk': False}, u'changed': False})
skipping: [localhost] => (item={'invocation': {'module_name': u'stat', 'module_args': ''}, 'item': 'slimer', u'stat': {u'uid': 1000, u'exists': True, u'woth': False, u'mtime': 1444959414.0001683, u'inode': 67109528, u'isgid': False, u'size': 4096, u'wgrp': False, u'isuid': False, u'isreg': False, u'pw_name': u'kragle', u'gid': 0, u'ischr': False, u'wusr': True, u'xoth': True, u'rusr': True, u'nlink': 7, u'issock': False, u'rgrp': True, u'gr_name': u'root', u'path': u'/opt/autodeploy/projects/slimer', u'xusr': True, u'atime': 1445344428.3137386, u'isdir': True, u'ctime': 1445344428.3157387, u'isblk': False, u'xgrp': True, u'dev': 64769, u'roth': True, u'isfifo': False, u'mode': u'0755', u'islnk': False}, u'changed': False})
skipping: [localhost] => (item={'invocation': {'module_name': u'stat', 'module_args': ''}, 'item': 'ansible-scaleio', u'stat': {u'uid': 1000, u'exists': True, u'woth': False, u'mtime': 1444959413.5571682, u'inode': 136953041, u'isgid': False, u'size': 4096, u'wgrp': False, u'isuid': False, u'isreg': False, u'pw_name': u'kragle', u'gid': 0, u'ischr': False, u'wusr': True, u'xoth': True, u'rusr': True, u'nlink': 5, u'issock': False, u'rgrp': True, u'gr_name': u'root', u'path': u'/opt/autodeploy/projects/ansible-scaleio', u'xusr': True, u'atime': 1445344428.3127387, u'isdir': True, u'ctime': 1445344428.3137386, u'isblk': False, u'xgrp': True, u'dev': 64769, u'roth': True, u'isfifo': False, u'mode': u'0755', u'islnk': False}, u'changed': False})

TASK: [verify-files | Check if source ISO files exist on autodeploynode] ******
ok: [localhost] => (item=hnl_mk_debug-image.2.0.1.iso)
ok: [localhost] => (item=rhel-server-7.1-x86_64-dvd.iso)
ok: [localhost] => (item=rhel-server-7.1-x86_64-dvd-hnl.iso)
ok: [localhost] => (item=rhel-guest-image-7.1-20150224.0.x86_64.qcow2)

TASK: [verify-files | Copy source ISO files to autodeploynode when offline] ***
skipping: [localhost] => (item={'invocation': {'module_name': u'stat', 'module_args': ''}, 'item': u'hnl_mk_debug-image.2.0.1.iso', u'stat': {u'uid': 1000, u'exists': True, u'woth': False, u'mtime': 1444939206.8611965, u'inode': 202124560, u'isgid': False, u'size': 54145024, u'wgrp': False, u'isuid': False, u'isreg': True, u'pw_name': u'kragle', u'gid': 0, u'ischr': False, u'wusr': True, u'xoth': False, u'rusr': True, u'nlink': 1, u'issock': False, u'rgrp': True, u'gr_name': u'root', u'path': u'/opt/autodeploy/resources/ISO/hnl_mk_debug-image.2.0.1.iso', u'xusr': False, u'atime': 1445355461.3078153, u'md5': u'b61423d24f4bda7e64b04163493fc0d6', u'isdir': False, u'ctime': 1445344428.3127387, u'isblk': False, u'xgrp': False, u'dev': 64769, u'roth': True, u'isfifo': False, u'mode': u'0644', u'checksum': u'4ed8350e7d332c9a8e65cd46041a2e5b72cf60b1', u'islnk': False}, u'changed': False})
skipping: [localhost] => (item={'invocation': {'module_name': u'stat', 'module_args': ''}, 'item': u'rhel-server-7.1-x86_64-dvd.iso', u'stat': {u'uid': 1000, u'exists': True, u'woth': False, u'mtime': 1444939178.3231938, u'inode': 202124558, u'isgid': False, u'size': 3890216960, u'wgrp': False, u'isuid': False, u'isreg': True, u'pw_name': u'kragle', u'gid': 0, u'ischr': False, u'wusr': True, u'xoth': False, u'rusr': True, u'nlink': 1, u'issock': False, u'rgrp': True, u'gr_name': u'root', u'path': u'/opt/autodeploy/resources/ISO/rhel-server-7.1-x86_64-dvd.iso', u'xusr': False, u'atime': 1445344623.6887577, u'md5': u'370662c78e36528bbaf3f2b901c29d6f', u'isdir': False, u'ctime': 1445344428.3127387, u'isblk': False, u'xgrp': False, u'dev': 64769, u'roth': True, u'isfifo': False, u'mode': u'0644', u'checksum': u'9af515956bb428f697aa6e30791c032a00060d1e', u'islnk': False}, u'changed': False})
skipping: [localhost] => (item={'invocation': {'module_name': u'stat', 'module_args': ''}, 'item': u'rhel-server-7.1-x86_64-dvd-hnl.iso', u'stat': {u'uid': 0, u'exists': True, u'woth': False, u'mtime': 1445346799.33697, u'inode': 202131250, u'isgid': False, u'size': 1212985344, u'wgrp': False, u'isuid': False, u'isreg': True, u'pw_name': u'root', u'gid': 0, u'ischr': False, u'wusr': True, u'xoth': False, u'rusr': True, u'nlink': 1, u'issock': False, u'rgrp': True, u'gr_name': u'root', u'path': u'/opt/autodeploy/resources/ISO/rhel-server-7.1-x86_64-dvd-hnl.iso', u'xusr': False, u'atime': 1445355476.3648167, u'md5': u'212e9e6832399b14a0511c8d2bf91cd0', u'isdir': False, u'ctime': 1445346799.33697, u'isblk': False, u'xgrp': False, u'dev': 64769, u'roth': True, u'isfifo': False, u'mode': u'0644', u'checksum': u'dbd8b6c329052cc60a199dd810f7040747729cff', u'islnk': False}, u'changed': False})
skipping: [localhost] => (item={'invocation': {'module_name': u'stat', 'module_args': ''}, 'item': u'rhel-guest-image-7.1-20150224.0.x86_64.qcow2', u'stat': {u'uid': 1000, u'exists': True, u'woth': False, u'mtime': 1444939203.785196, u'inode': 202124559, u'isgid': False, u'size': 425956864, u'wgrp': False, u'isuid': False, u'isreg': True, u'pw_name': u'kragle', u'gid': 0, u'ischr': False, u'wusr': True, u'xoth': False, u'rusr': True, u'nlink': 1, u'issock': False, u'rgrp': True, u'gr_name': u'root', u'path': u'/opt/autodeploy/resources/ISO/rhel-guest-image-7.1-20150224.0.x86_64.qcow2', u'xusr': False, u'atime': 1445355481.042817, u'md5': u'b068d0e9531699516174a436bf2c300c', u'isdir': False, u'ctime': 1445344428.3127387, u'isblk': False, u'xgrp': False, u'dev': 64769, u'roth': True, u'isfifo': False, u'mode': u'0644', u'checksum': u'5f3d607e7a490ae99dbdbb85ae979122ed5567fb', u'islnk': False}, u'changed': False})

TASK: [verify-files | Check if source docker files exist on autodeploynode] ***
ok: [localhost] => (item=cscdock_hanlon.tar)
ok: [localhost] => (item=cscdock_atftpd.tar)
ok: [localhost] => (item=mongo.tar)

TASK: [verify-files | Copy source docker files to autodeploynode when offline] ***
skipping: [localhost] => (item={'invocation': {'module_name': u'stat', 'module_args': ''}, 'item': 'cscdock_hanlon.tar', u'stat': {u'uid': 1000, u'exists': True, u'woth': False, u'mtime': 1444939284.045204, u'inode': 2344562, u'isgid': False, u'size': 737451520, u'wgrp': False, u'isuid': False, u'isreg': True, u'pw_name': u'kragle', u'gid': 0, u'ischr': False, u'wusr': True, u'xoth': False, u'rusr': True, u'nlink': 1, u'issock': False, u'rgrp': True, u'gr_name': u'root', u'path': u'/opt/autodeploy/resources/docker/cscdock_hanlon.tar', u'xusr': False, u'atime': 1445355482.7408173, u'md5': u'e15e5307a59ac256d86e08fd2d627049', u'isdir': False, u'ctime': 1445344428.3127387, u'isblk': False, u'xgrp': False, u'dev': 64769, u'roth': True, u'isfifo': False, u'mode': u'0644', u'checksum': u'aa6eade25f96eaffff5fcd3610b99a5bcc0671bc', u'islnk': False}, u'changed': False})
skipping: [localhost] => (item={'invocation': {'module_name': u'stat', 'module_args': ''}, 'item': 'cscdock_atftpd.tar', u'stat': {u'uid': 1000, u'exists': True, u'woth': False, u'mtime': 1444939283.564204, u'inode': 2344560, u'isgid': False, u'size': 210205184, u'wgrp': False, u'isuid': False, u'isreg': True, u'pw_name': u'kragle', u'gid': 0, u'ischr': False, u'wusr': True, u'xoth': False, u'rusr': True, u'nlink': 1, u'issock': False, u'rgrp': True, u'gr_name': u'root', u'path': u'/opt/autodeploy/resources/docker/cscdock_atftpd.tar', u'xusr': False, u'atime': 1445355485.5838177, u'md5': u'cbec0031f66d6740dfd906040d732176', u'isdir': False, u'ctime': 1445344428.3127387, u'isblk': False, u'xgrp': False, u'dev': 64769, u'roth': True, u'isfifo': False, u'mode': u'0644', u'checksum': u'6178c010cd2190ce118f7cb91c3e6263adf7c8ad', u'islnk': False}, u'changed': False})
skipping: [localhost] => (item={'invocation': {'module_name': u'stat', 'module_args': ''}, 'item': 'mongo.tar', u'stat': {u'uid': 1000, u'exists': True, u'woth': False, u'mtime': 1445018430.2809272, u'inode': 2344561, u'isgid': False, u'size': 267695616, u'wgrp': False, u'isuid': False, u'isreg': True, u'pw_name': u'kragle', u'gid': 0, u'ischr': False, u'wusr': True, u'xoth': False, u'rusr': True, u'nlink': 1, u'issock': False, u'rgrp': True, u'gr_name': u'root', u'path': u'/opt/autodeploy/resources/docker/mongo.tar', u'xusr': False, u'atime': 1445355486.4398177, u'md5': u'd9488bf99a1a04437bea0f64e884c7dd', u'isdir': False, u'ctime': 1445344428.3127387, u'isblk': False, u'xgrp': False, u'dev': 64769, u'roth': True, u'isfifo': False, u'mode': u'0644', u'checksum': u'4466db0a806dad533aa48ed9b1fea796f2e0f5ea', u'islnk': False}, u'changed': False})

PLAY RECAP ********************************************************************
localhost                  : ok=36   changed=8    unreachable=0    failed=0

adean20csc commented:
Fixed PR40 and cleaned up lab variables

jcpowermac commented on ansible/roles/dhcp-server/templates/dhcpd.conf.j2:
```diff
@@ -17,10 +16,10 @@ subnet {{ pxe_subnet_name }} netmask {{ pxe_subnet_mask }} {
     option routers              {{ pxe_gateway }};
     option subnet-mask          {{ pxe_subnet_mask }};
-    option domain-name          "{{ domain_name }}";
+    option domain-name          {{ domain_name }};
```
@adean20csc those values within dhcpd.conf must be quoted.

jcpowermac commented on ansible/roles/dhcp-server/templates/dhcpd.conf.j2:
```diff
@@ -17,10 +16,10 @@ subnet {{ pxe_subnet_name }} netmask {{ pxe_subnet_mask }} {
     option domain-name-servers  {{ dns1 }};
     next-server                 {{ auto_deploy_node }};
-    option tftp-server-name     "{{ auto_deploy_node }}";
+    option tftp-server-name     {{ auto_deploy_node }};
```
same here, the quotes need to be here.

jcpowermac commented:
Code Review :+1:

mtnbikenc commented:
Code review :+1:
